### PR TITLE
UI Cockpit

### PR DESCRIPTION
**Pull request by nhaagen**
Proposal for the Cockpit,  aka UIRevision-tabbar, as described in http://www.ilias.de/docu/goto.php?target=wiki_1357_Page_Layout_Revision.

### CODE REVIEW

**Comment by klees** [comment]
Hi @Amstutz,

we started to build PRs for the Page Layout to have a place to explain the Notification Center and the Cockpit (as we have called it now).

Best regards!

File: src/UI/Component/Layout/Page/Factory.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     *     The menubar consists of two parts: the main entries in form of a
+     *     button-collection and a section ("slate") that opens upon the selection
+     *     of one of them, very similar to a fly-out or mega menu.
+     *
+     *   effect: >
+     *     The menubar is always visible and available (except in exam/kiosk mode).
+     *     In a desktop environment, a vertical bar is rendered on the left side
+     *     of the screen covering the full height. Entries are aligned vertically.
+     *
+     *     For mobile devices, the menubar is rendered horizontally on the bottom
```
Comment (catkrahl) [inline src/UI/Component/Layout/Page/Factory.php]: at the bottom

File: src/UI/Factory.php [hunk 2]
```diff
@@ -336,4 +336,15 @@ public function panel();
 	 **/
 	public function modal();
 
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: Layout elements
```
Comment (Amstutz) [inline src/UI/Factory.php]: Please provide a better description here. What is a layout, what is not a layout? Is a deck a layout?

File: src/UI/Component/Layout/Factory.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * This is what a factory for layouts looks like.
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     A page-element describes a section or subsection of the ILIAS Application;
+	 *     the page thus is the total view upon ILIAS.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Factory.php]: What is a page element? Should it not be a page layout?

File: src/UI/Component/Layout/Factory.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	 *
+	 *
+	 * ----
+	 *
+	 * @return  \ILIAS\UI\Component\Layout\Page\Factory
+	 */
+	public function page();
```
Comment (Amstutz) [inline src/UI/Component/Layout/Factory.php]: Why page? Why not section? Section Layout? We are unsure.

File: src/UI/Component/Layout/Page/Factory.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+     *   composition:
+     *     1: The cockpit MUST contain a bar.
+     *     2: The cockpit SHOULD contain a slate.
+     *
+     * ----
+     *
+     * @return  \ILIAS\UI\Component\Layout\Page\Cockpit\Factory
+     */
+    public function cockpit();
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: This would currently read as Cockpit Page Layout. Would not Cockpit Section Layout be more appropriate?

File: src/UI/Component/Layout/Page/Factory.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+ * This is the factory for page-layout components
+ */
+interface Factory {
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The cockpit is a unique page section that holds globally available
+     *     tools and navigational elements.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: Currently it reads as Page Layout (Cockpit Page Layout). Consider to revise naming.

File: src/UI/Component/Layout/Page/Factory.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * This is the factory for page-layout components
+ */
+interface Factory {
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The cockpit is a unique page section that holds globally available
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: Globally available tools? That would leave out tools and structures only available in certain contexts (such as the page editor or learning module). E.g. with this description, some navigation for the learning module would not belong here.

File: src/UI/Component/Layout/Page/Factory.php [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+     *     tools and navigational elements.
+     *     It bundles access to navigational strategies (search or the repository tree),
+     *     tools parallel to the content like the help, and services unrelated to
+     *     the actual content like the user's profile or desktop.
+     *
+     *   composition: >
+     *     The cockpit is built of a bar holding one to five entries, which will
+     *     usually open the "slate", a larger area next to the bar providing room
+     *     for a large variety of components.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: Five might be the default. But it might be configured to hold more. Further the section next to the bar also belongs to the cockpit correct? According the current description, this section can not contain content that changes depending on the context. I could imagine however, that e.g. the navigation for a learning module could be displayed in there. Also the navigation for a course with the course itself as root, would be neatly placed here (inside the slate).

File: src/UI/Component/Layout/Page/Factory.php [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+     *
+     *   effect: >
+     *     The cockpit(-bar) is always visible and available (except in exam/kiosk mode).
+     *     It is _the_ main navigational item apart from links/buttons in the content.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: Maybe also add breadcrumb here.

File: src/UI/Component/Layout/Page/Factory.php [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+     *
+     *   rivals:
+     *     tab bar: >
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: Capitalize.

File: src/UI/Component/Layout/Page/Factory.php [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+     *       The cockpit shall not be used to substitute object-actions;
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: Please elaborate what is meant by object-actions. Add New item? Add Comment? Display Comments? Would the navigation of a learning module be considered a content related action?

File: src/UI/Component/Layout/Page/Factory.php [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+     *       displays further information at first glance (i.e. the calendar).
+     *       The cockpit may reference those tools as well, but rather in form
+     *       of a link than a widget.
+     *
+     *
+     * rules:
+     *   usage:
+     *     1: There MUST be but one cockpit on the page.
+     *
+     *   interaction:
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: delete if empty.

File: src/UI/Component/Layout/Page/Factory.php [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+
+     *   composition:
+     *     1: The cockpit MUST contain a bar.
+     *     2: The cockpit SHOULD contain a slate.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: What about empty cockpits (configured to be empty). E.g. for the anonymous user?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+namespace ILIAS\UI\Component\Layout\Page\Cockpit;
+/**
+ * This is the factory for cockpit-components
+ */
+interface Factory {
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The bar is the primary visual manifestation of the cockpit.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Not sure if this is helpful, maybe rephrase to "The bar is the part of the cockpit that remains visible and therefore accessible at any time".

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   composition: >
+     *     The bar is the cockpit's contanier for entries.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: How many? Are they configurable? Can/Do they change upon changing the context?

File: src/UI/Component/Layout/Page/Factory.php [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+     *   rivals:
+     *     tab bar: >
+     *       The cockpit shall not be used to substitute object-actions;
+     *       those remain in the tab-bar.
+     *     Personal Desktop: >
+     *       The Personal Desktop provides access to services and tools and
+     *       displays further information at first glance (i.e. the calendar).
+     *       The cockpit may reference those tools as well, but rather in form
+     *       of a link than a widget.
+     *
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php]: The Notification center is a further important rival. When is something placed there. When is something placed here?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   effect: >
+     *     In a desktop environment, a vertical bar is rendered on the left side
+     *     of the screen covering the full height minus the header area.
+     *     Entries are aligned vertically.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Does it extend above the footer? Those it contain scrollbars if there are more items than can be displayed?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *     For mobile devices, the bar is rendered horizontally on the bottom
+     *     of the screen with the entries aligned horizontally.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Some questions as a few lines above.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *     The cockpit(-bar) is always visible and available (except in exam/kiosk mode).
+     *
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Please elaborate in more details, what happens if there are more than five items (or four in mobile view) stuffed in here. Are they moved into the "More" Section. Maybe also describe the special more section here? How is this used? What is put in there.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+     * rules:
+     *   usage:
+     *     1: There MUST be but one cockpit bar on the page.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: This is somewhat redundant to what is already written in the rules for the cockpit. Maybe leave out here. Maybe this is not even correct. What if there are no items (e.g. anonymous user)?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   composition:
+     *     1: The bar MUST NOT contain items other than entries.
+     *     2: The bar MUST contain at least one entry.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Why? How can we ensure such a thing?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+     *     3: The bar MUST NOT contain more than five entries.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: What happens if there are more? Please describe in effect? Could there be a ways to configure more?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   style:
+     *     1: The bar MUST have a fixed witdth (desktop).
+     *     2: The bar MUST have a fixed height (mobile).
+     *
+     * ----
+     * @param  entry[] $entries
+     * @return  \ILIAS\UI\Component\Layout\Page\Cockpit\Bar
+     */
+    public function bar($entry_1, $entry_2=null, $entry_3=null, $entry_4=null, $entry_5=null);
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: This seems odd. Can you not stuff more than five in there that would go into the more section? Would you really hardly constrict that to five? I would rather make this a recommendation and leave the final decision to the admin. Note, this will be configurable through the administration.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     An entry in this context describes an item in the cockpit-bar.
+     *     Usually, it is a labeled, clickable icon which triggers a slate.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: It is always labeled and usually contains an icon. Move this to composition.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 25]
```diff
@@ -0,0 +1,121 @@
...
+     *     Alternatively, the link directly targets a page.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Move this and the later part of the above to effect. "Upon click, the entry triggers a slate or links directly to some page".

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 26]
```diff
@@ -0,0 +1,121 @@
...
+     * @return  \ILIAS\UI\Component\Layout\Page\Cockpit\Bar
+     */
+    public function bar($entry_1, $entry_2=null, $entry_3=null, $entry_4=null, $entry_5=null);
+
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     An entry in this context describes an item in the cockpit-bar.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: They immediately offer access to the fundamental functionality of ILIAS, such as navigation to some parts of the repository or links to the aspects of the personal desktop.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 27]
```diff
@@ -0,0 +1,121 @@
...
+     *     Usually, it is a labeled, clickable icon which triggers a slate.
+     *     Alternatively, the link directly targets a page.
+     *
+     *   composition: >
+     *     Entries consist of a caption, an icon and, optionally, a slate.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Are you sure this is correct. Are slates not linked to an entry? Can entries not be put into other entries? What about the "More" entry? Can this entry contain other entry?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 28]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     * rules:
+     *   usage:
+     *     1: Entries MUST only be inside a cockpit-bar.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: What about the more button?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 29]
```diff
@@ -0,0 +1,121 @@
...
+     *     2: Entries CAN be connected with a slate
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Use MAY not CAN. Above you write, that entries can be composed of slates. This is somewhat contradictive.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 30]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   interaction:
+     *     1: Entries MUST have an active-state.
+     *     2: Entries CAN trigger a slate.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: MAY

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 31]
```diff
@@ -0,0 +1,121 @@
...
+     *     3: Entries CAN trigger navigation to a content element.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: MAY

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 32]
```diff
@@ -0,0 +1,121 @@
...
+     *     4: Entries CAN be unclickable.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Really? Why should that be? In this case, they should not be displayed.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 33]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   composition:
+     *     1: Entries MUST have a caption.
+     *     2: Entries MUST have a graphical representation like an icon or glyph.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: I would make this a SHOULD.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 34]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   style:
+     *     1: Graphical elements of entries SHOULD be of the same size.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Make this a MUST.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 35]
```diff
@@ -0,0 +1,121 @@
...
+     * @return  \ILIAS\UI\Component\Layout\Page\Cockpit\Entry
+     */
+    public function entry($caption, $icon, $action);
+
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The slate is a page-area within the cockpit; it acts like an enhanced
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Maybe rephrase page-area to section.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 36]
```diff
@@ -0,0 +1,121 @@
...
+     *     fly-out menu for cockpit-entries.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Here we must be more specific. What can be done inside such slates and what not? Can you perform little workflows (such as hand in an exercise, leave a comment or change the attributes of an element added to the page by using the page editor, perform a search, use as filter for the content)? Can we narrow this description down somewhat?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 37]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   composition: >
+     *     The slate can hold a large variety of components. These can be (further)
+     *     navigational entries, forms, text and images or combinations of those.
+     *     For example, a submenu could be expressed by (accordion-)panels with
+     *     underlying buttons.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Do we really want full blown forms here? For what reason?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 38]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *   effect: >
+     *     When triggered, the slate opens on the right hand of the cockpit-bar,
+     *     between bar and content, thus "pushing" the content to the right.
+     *     The slate will allways have a "close"-button at its bottom.
+     *
+     *     The contents of a slate can vary heavily:
+     *     A search form, the repository tree, contextual help,
+     *     further navigation via buttons, etc.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: The content is described under "Composition".

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 39]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     *     When content-length exceeds the slate's height, the area above the
+     *     close button will start scrolling.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Where is the scrollbar placed? How will it scroll?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php (hunk 34 above)
Comment (Amstutz) [reply src/UI/Component/Layout/Page/Cockpit/Factory.php]: Please also an accessability rule here. E.g. "Every entry MUST be accessible by only using the keyboard".

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 40]
```diff
@@ -0,0 +1,121 @@
...
+     *
+     * rules:
+     *   usage:
+     *     1: There MUST be only one slate visible on the page.
+     *
+     *   accessibility:
+     *     1: The slate MUST be triggered by an entry.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: Why is this accessibility? We should add a generic accessibility rule here, such as: "Actions or navigational elements offered inside a slate, MUST be accessible by keyboard.

Comment (Amstutz) [reply src/UI/Component/Layout/Page/Cockpit/Factory.php]: Further: "Slates MUST be closable by only using the keyboard.".

File: src/UI/Component/Layout/Page/Cockpit/Factory.php [hunk 41]
```diff
@@ -0,0 +1,121 @@
...
+     *   composition: >
+     *     Entries consist of a caption, an icon and, optionally, a slate.
+     *
+     * rules:
+     *   usage:
+     *     1: Entries MUST only be inside a cockpit-bar.
+     *     2: Entries CAN be connected with a slate
+     *
+     *   interaction:
+     *     1: Entries MUST have an active-state.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Cockpit/Factory.php]: +1. Please add some indication, at which point this active-state should be active. This can be quite hard. E.g. which (if any) entry is active by looking at some blog inside the repository. Which is active while looking at the administration? Which is active while looking at the PD? Are they only active if the slate is open?

File: src/UI/Component/Layout/Page/Cockpit/Factory.php (hunk 39 above)
Comment (Amstutz) [reply src/UI/Component/Layout/Page/Cockpit/Factory.php]: What is about the hight of a slate?

**Review by Amstutz — CHANGES_REQUESTED** [review]
This PR seems incomplete. Please add the complete interface. Also, add the mockups for a quick overview for other developers.

There are some key aspects that need to be discussed:
- The number of items, is it really fixed?
- Can it be configured, that now items are to be displayed?
- What can be done inside slates? What not?

File: src/UI/Component/Layout/Page/Factory.php (hunk 7 above)
Comment (nhaagen) [reply src/UI/Component/Layout/Page/Factory.php]: We switched to "navigational elements and globally available tools" and excluded modification.

File: src/UI/Component/Layout/Page/Cockpit/Factory.php (hunk 15 above)
Comment (nhaagen) [reply src/UI/Component/Layout/Page/Cockpit/Factory.php]: >Are they configurable?
yes, but this is not really a concern of UI?

Comment (Amstutz) [reply src/UI/Component/Layout/Page/Cockpit/Factory.php]: > yes, but this is not really a concern of UI?

It is to the extend to which the UI-Component and the according rules have to be flexible enough to handle all possible configuration. The configuration mechanics are not part of this though.

**Comment by klees** [comment]
Hi @Amstutz,
there were some comments in your review revolving around the amount of entries in the bar and the "more"-entry, that I want to answer in bulk. It is also be connected to the question about configurability, that also appeared in your comments.
In our current understanding there is no special "more"-entry from the perspective of the UI-framework. The "more"-entry is just on standard entry, opening a slate. The rules how and when entries in the bar should be moved to the "more"-entry to us needs to be solved in some "cockpit"-service (superseeding ilMainMenuGUI probably), that also answers the question about which entries are shown to whom, how one could configure entries etc.
This is connected to the question why we encoded exactly five entries in the `bar`-constructor. The general idea (as far as I understand) of the bar is, that the user has a fixed amount of entries he could use for navigation and the tools we tried to describe. These entries should be the same on desktop and on mobile. The entries furthermore should be visible all the time to give the user some fixed points to return to if he is somehow lost in navigation. This, the limited amount of space on mobile devices and the stipulation that "more" is not a special entry just puts the maximum of five (or arguably six) to the amount of entries in the bar.
Does this make sense somehow?
Best regards!

**Review by alex40724 — CHANGES_REQUESTED** [review]
I just realised that you are already started doing PRs ins this. I still disaprove the approach. I do not have a feeling that any of my concerns have been adressed. I will continue to list my arguments on the FW page.

**Comment by klees** [comment]
Hi @alex40724,
that is not the case at all. One of your concerns was that the separation between the "Notification Center" and the "Tab Bar" is not clear conceptually. This is exactly the reason we started to write down the concepts in this PRs. This is also exactly the form we agreed upon to discuss UI-concepts, after all.
Your entry in the Feature Wiki Page was answered by YS as well.
Best regards!

**Comment by Amstutz** [comment]
@alex40724 This PR was devised to line out the concept more clearly. The FR is somehow limited to tackle the level of detail we felt is needed to discuss our proposal in depth. Note that we do not mean to replace the FR and the discussion there, but enhance it with technical and semantical details.

FR: Describes the general idea of a page layout revision along with the necessary mockups. You did right in commenting there with issues tackling the complete concept.
PRs: The PR's you see here describe several elements in more detail. They are necessary, since we plan to introduce a whole bunch of new UI Concepts. Obviously they are not yet accepted and will be obsolete if the feature is rejected as a whole.

Since your concerns are mainly about the complete concept and not about semantical or technical details, I will answer to them in the according FR (for all less involved readers, we are talking about http://www.ilias.de/docu/goto_docu_wiki_1357_Page_Layout_Revision.html).

**Comment by alex40724** [comment]
Hi @klees,
we might have different ideas on how to proceed here
- YS  only commented on my alternative proposal. Not on my initial concerns on this proposal. I have no idea how scalability, configurabiliy or focusing on important tasks will work right now.
- I can't see anything in this PR that does help me with my concerns, sorry.
- I definitely thought to have some alternatives mockups that adress some of these issues before discussing technical interfaces.

@Amstutz: 
> PRs: The PR's you see here describe several elements in more detail. They are necessary, since we plan to introduce a whole bunch of new UI Concepts. Obviously they are not yet accepted and will be obsolete if the feature is rejected as a whole.

But why putting so much effort into technical details before agreeing on the general idea and risking this work is for nothing?

**Comment by Amstutz** [comment]
@alex40724 To be honest, I was glad to have the PR's to understand our own proposal better. I think it is good to know early in the discussion to really know what we are talking about. If you read my comments, you see, that we discovered several shallows while discussing this PR (e.g. what is a sidebar aka cockpit, especially what can the sidebar aka cockpit contain and what not, what should be performable there and what note). 

You are right, we took quite some risk of going to such detail early in the process. However, it is a risk I am willing to take, to offer as much clarity as early as possible. Note that this level of detail and discussion would completely blow the FR.

**Comment by klees** [comment]
Hi @alex40724,
I'm sorry that you cannot see anything that helps you in this PR now. I will be happy to point you to the according locations once we are finished. This PRs to me in fact help to put the conceptual considerations in a form we all know and agreed upon and explore the idea, like Timon mentioned. This most probably will answer your question how to decide what belongs to the tab bar vs. the notification center and more. We try to work as transparently and openly as possible here, this is not the final state of this PR. We'll also include some mockups.
Best regards!

**Comment by klees** [comment]
We will open new PRs once our components are ready. Thx for Feedback!
